### PR TITLE
Fix scaling in SplitContainer controls for high DPI machines

### LINKED ISSUE #3168: Controls (with Anchor Right) scaling bugs inside splitContainer
* .NET Core Version: **3.1.3**

* Have you experienced this same bug with .NET Framework?: **Yes**

**Problem description:**

Controls, that placed inside splitContainer and have in Anchor  property `Right` (may be `Bottom` too) flag make their size and position wrong with other dpi or font.

Designer (Win7 100% dpi):
Designer

Win7 auto Font 100% dpi (problems with core due to default font changed):
W7_100_AF

Win7 auto Font 100% and 125% dpi (problems due to dpi change). No core here because with auto font it's mess even with 100% dpi:
W7_100+125_AF

Win7 fixed Font 100% dpi (all ok):
W7_100_DF

Win7 fixed Font 125% dpi (problems due to dpi change):
W7_125_FF_net
W7_125_FF_core

All screens from attached project.
This bug only valid when dpi > 100% (system font size?), so all screenshot where you see _font set in designer_ - not affected, because of 100 dpi.

Some notes:

Removing fixed panel from splitContainer lead to different behavior (still wrong).

Due to WinForms designer/dpi scaling logic bug, font for the form must be set manually - in `Form1.Designer.cs` **before** `AutoScaleDimensions` set, or in form constructor **before** `InitializeComponent();`
Designer puts font settings **after** scale logic, **that completely break it**:
Snipaste_2020-04-30_19-18-56
I have opened a [new issue](https://github.com/dotnet/winforms/issues/3202) for this.

[Blog post](http://blog.davidegrayson.com/2013/04/unusable-ui-caused-by-windows-forms.html) about scaling problems with `Anchor Right/Bottom`.

**Expected behavior:**
Same position and proportional size change.

**Minimal repro:** 
[WinFormsCoreTest_dpi.zip](https://github.com/dotnet/winforms/files/4557769/WinFormsCoreTest_dpi.zip)

------------------------------------------------------------------------------------------

And for last - Seems a regression in Win10 .net runtime? On win 7 working good both on .net and core, on Win10 only on core.
W10_100_DF
On win10 I can't achieve normal result even with 100% dpi (no matter auto font, or manual set).
I understand that this is wrong place for .net bugs, but may be you transfer it to .net team?
Win10 pro 1809/1909.

### PR DESCRIPTION
**Pull request by dreddy-work**
Fixes #3168 in combination with https://github.com/dotnet/winforms/pull/7956

## Proposed changes

- Scaling SplitterDistance property when DPI changed event occurs.
- Updating Panel sizes when FixedPanel property is set and DPI changed event occurs.
- Add unit test for future reference.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7990)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/SplitContainerTests.cs [hunk 1]
```diff
@@ -20,5 +22,38 @@ public void SplitContainer_Constructor()
...
+            using var splitContainer = new SplitContainer();
+
+            splitContainer.FixedPanel = FixedPanel.Panel1;
+            splitContainer.Location = new Drawing.Point(0, 0);
+            splitContainer.Margin = new Padding(0);
+            splitContainer.Name = "splitContainer2";
+            splitContainer.Orientation = Orientation.Horizontal;
+            splitContainer.Size = new Drawing.Size(812, 619);
+            splitContainer.SplitterDistance = 90;
+            splitContainer.SplitterWidth = 2;
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/SplitContainerTests.cs:47]:
```suggestion
            using var splitContainer = new()
            {
                FixedPanel = FixedPanel.Panel1,
                Location = new(0, 0),
```

File: src/System.Windows.Forms/tests/UnitTests/SplitContainerTests.cs [hunk 2]
```diff
@@ -20,5 +22,38 @@ public void SplitContainer_Constructor()
...
+        [InlineData(3.5 * DpiHelper.LogicalDpi)]
+        public void SplitContainer_Properties_Scaling(int newDpi)
+        {
+            // Run tests only on Windows 10 versions that support thread dpi awareness.
+            if (!PlatformDetection.IsWindows10Version1803OrGreater)
+            {
+                return;
+            }
+
+            using var form = new Form();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/SplitContainerTests.cs:37]: Why is the form needed? The splitter isn't hosted on the form.

**Review by RussKie — CHANGES_REQUESTED** [review]

File: src/System.Windows.Forms/tests/UnitTests/SplitContainerTests.cs [hunk 3]
```diff
@@ -20,5 +22,38 @@ public void SplitContainer_Constructor()
...
+            using var splitContainer = new SplitContainer();
+
+            splitContainer.FixedPanel = FixedPanel.Panel1;
+            splitContainer.Location = new Drawing.Point(0, 0);
+            splitContainer.Margin = new Padding(0);
+            splitContainer.Name = "splitContainer2";
+            splitContainer.Orientation = Orientation.Horizontal;
```
[the remark this answers is not in the document]
Comment (dreddy-work) [reply src/System.Windows.Forms/tests/UnitTests/SplitContainerTests.cs:44]: Currently we have a known issue of not being able to set process Dpi on VMs. this test is for future reference but is not being executed. Not planning on adding more tests at this point as they are not going to be executed for DPI. Happy to follow up later.

[the remark this answers is not in the document]
Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/SplitContainerTests.cs]: Ah, right. Didn't notice the skip parmeter.

**Review by RussKie — DISMISSED** [review]

**Review by RussKie — APPROVED** [review]

**Pull request merged**